lib/modules/datasource/common.ts

```diff
@@ -181,14 +181,36 @@ export function applyConstraintsFiltering<
         continue;
       }
 
-      const satisfiesConstraints = constraint.some(
-        // If the constraint value is a subset of any release's constraints, then it's OK
-        // fallback to release's constraint match if subset is not supported by versioning
-        (releaseConstraint) =>
-          !releaseConstraint ||
-          (versioning.subset?.(configConstraint, releaseConstraint) ??
-            versioning.matches(configConstraint, releaseConstraint)),
-      );
+      let satisfiesConstraints = false;
+      for (const releaseConstraint of constraint) {
+        if (!releaseConstraint) {
```

Comment:
we can have `undefined` in our `constraint` array? 😕 we should log a message (maybe debug once)?